src/cards/entity-card/entity-card.ts

```diff
@@ -122,13 +123,14 @@ export class EntityCard extends LitElement implements LovelaceCard {
                         hasHold: hasAction(this._config.hold_action),
                     })}
                     .hide_info=${primary == null && secondary == null}
+                    .hide_icon=${hideIcon}
                 >
-                    <mushroom-shape-icon
+                    ${!hideIcon ? html`<mushroom-shape-icon
```

Comment:
```suggestion
                    ${!icon ? html`<mushroom-shape-icon
```
Maybe we can hide icon when no icon is defined in the template like for primary and secondary option 🙂

Reply:
Yes, I was thinking about this. Maybe changing how `hideIcon` is initialiazed gives a better consistancy :)